portfolio/src/main/webapp/game.js

```diff
@@ -9,7 +9,23 @@ class Board {
       [0, 0, 0]
     ];
     this.drawGrid();
-    var red = true; 
+    this.red = true; 
+    this.numTiles = 0;
```

Comment:
What is this info used for?

Reply:
Num tiles is used to check if the board is full in O(1). It increases after every valid move. It's compared against the size of the board in my isFull() function.